src/global/ValueId.h

```diff
@@ -262,9 +262,26 @@ class ValueId {
     return addDatatypeBits(bits, Datatype::Bool);
   }
 
+  /// Create a `ValueId` for a boolean value, represented as 0/1 instead of
+  /// false/true.
+  static constexpr ValueId makeFromBinaryBool(bool b) noexcept {
+    auto bits = static_cast<T>(b);
+    bits |= static_cast<T>(true) << 1;
+    return addDatatypeBits(bits, Datatype::Bool);
+  }
+
   // Obtain the boolean value.
   [[nodiscard]] bool getBool() const noexcept {
-    return static_cast<bool>(removeDatatypeBits(_bits));
+    return static_cast<bool>(removeDatatypeBits(_bits) & 1);
+  }
+
+  // Obtain the boolean value as a string view.
+  std::string_view getBoolLiteral() const noexcept {
+    bool value = getBool();
+    if (removeDatatypeBits(_bits) & 0b10) {
```

Comment:
The `removeDatatypeBits` seems unnecessary, as you have a bitiwse AND which zeroes them out anyway.